webui/src/pages/repositories/repository/fileRenderers/data.tsx

```diff
@@ -26,24 +28,52 @@ LIMIT 20`
 FROM read_csv(lakefs_object('${repoId}', '${refId}', '${path}'), DELIM='\t', AUTO_DETECT=TRUE) 
 LIMIT 20`
     }
-    const [query, setQuery] = useState<string>(initialQuery)
     const [shouldSubmit, setShouldSubmit] = useState<boolean>(true)
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
     const [data, setData] = useState<any[] | null>(null);
     const [error, setError] = useState<string | null>(null)
+    const [loading, setLoading] = useState<boolean>(false)
+
+    const sql = useRef<HTMLTextAreaElement>(null);
+
+    const handleSubmit = useCallback((event: FormEvent<HTMLFormElement>) => {
+        event.preventDefault()
+        setShouldSubmit(!shouldSubmit)
+    }, [setShouldSubmit])
 
     useEffect(() => {
+        setLoading(true)
         withConnection(async conn => {
-            const results = await conn.query(query)
+            if (!sql || !sql.current) return
+            const results = await conn.query(sql.current.value)
             const data = results.toArray()
             setData(data)
             setError(null)
+
         }).catch(e => {
```

Comment:
You're mixing paradigms. `.catch` is Promise syntax and `async-await` is a different approach. If you're using `async-await`, you should use plain old `try-catch-finally`. That's the idea of `async-await` - writing sync-style code, and abstracting away all of the Promise handling, continuation, error handling, etc.

Reply:
Fixed